src/sage/modular/quasimodform/element.py

```diff
@@ -728,3 +730,72 @@ def derivative(self):
         hom_comp = self.homogeneous_components()
 
         return sum(f.serre_derivative() + R(k) * u * f * E2 for k, f in hom_comp.items())
+
+    def _compute(self, X):
+        r"""
+        Compute the coefficients of `q^n` of the power series of self,
+        for `n` in the list `X`.  The results are not cached.  (Use
+        coefficients for cached results).
+
+        EXAMPLES::
+
+            sage: E2 = QuasiModularForms(1).0
+            sage: E2.q_expansion(10)
+            1 - 24*q - 72*q^2 - 96*q^3 - 168*q^4 - 144*q^5 - 288*q^6 - 192*q^7 - 360*q^8 - 312*q^9 + O(q^10)
+            sage: E2._compute([3, 6])
+            [-96, -288]
+            sage: E2._compute([])
+            []
+        """
+        if not isinstance(X, list) or not X:
+            return []
+        bound = max(X)
+        q_exp = self.q_expansion(bound + 1)
+        return [q_exp[i] for i in X]        
+
+    def coefficients(self, X):
+        r"""
+        The coefficients a_n of self, for integers n>=0 in the list
+        X. If X is an Integer, return coefficients for indices from 1
+        to X.
```

Comment:
Same as above, this block should be of the form:
```
Return blah blah... (one sentence description)
[blank line]
Longer description IF needed.
```
Note that the documentation of the code you mostly copy/pasted is quite old and not very up-to-date!
Also it is not recommended to use "self" in the doc, you should use "this graded quasimodular form".